vtex-apps/return-app
# V3 store rma details

#### What problem is this solving?

This PR brings the Return Request Details page to the store. It reuses the components created for the admin side.
Most of the changes (51 files changing 😱) is due to bringing the modules from `react/admin` to `react/common` and removing the prefix `admin/` for the shared messages. The files that really need a review are:

__Back end__

`node/resolvers/ReturnRequestResponse.ts` and `graphql/types/ReturnRequest.graphql`
I changed the resolver for `ReturnRequestResponse` so when a request comes from the store we hide the field `subimittedBy`. I had to change the GraphqQL schema in order to reflect that.

__Front end__

`react/admin/hooks/useUpdateRequestStatus.ts`, `react/admin/AdminReturnDetails.tsx`, `react/admin/provider/UpdateRequestStatusProvider.tsx` and `react/common/provider/ReturnDetailsProvider.tsx`

I decoupled the Provider responsible for the return request details. Before, it exposed the data from return request details and the function to update it. Since the update functionality is only needed in the admin side, I moved it to its own provider. This way, the function is not available in the store side.

`react/common/components/RenderStatus.tsx` and `react/common/components/returnList/ListTableSchema.tsx`

I decoupled the function to render the status to its own component so we can reuse it in other components.

`react/store/ReturnRequest/StoreReturnDetails.tsx`

The main entry point for the store return details page. It's wrapped in the context to have access to data from `returnRequest` resolver.

`react/store/ReturnRequest/loaders/StoreReturnDetailsLoader.tsx`

The loader for the return request details page on the store side.

`react/common/components/ReturnDetails/OrderLink.tsx` and `react/common/components/ReturnDetails/CurrentRequestStatus.tsx`

Two new components to the return request details page. One that is a link to the order page (both admin and store) and another one that shows the current status.

#### How to test it?

The branch is linked in this [workspace](https://v3rmastoredetails--powerplanet.myvtex.com). Logged as a store user, one can visit the page to see the return request details.

#### Screenshots or example usage:
![Screenshot 2022-07-06 at 15-49-23 Tienda online de smartphones tablets e informática](https://user-images.githubusercontent.com/38737958/177776616-1bfcc419-3a30-4b4b-9078-4a53f6184977.png)
